### PR TITLE
Add OSD 9 positions, Fix High-CPU Load

### PR DESCRIPTION
**Pull request by raven2cz**
OSD position: bottom-right (default), top-right and center support.

I used components which were already included in the project only. Most simple way as possible.

### CODE REVIEW

File: volctl/prefs.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def _setup_ui(self):
         hbox = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL)
         row.add(hbox)
         label = Gtk.Label(xalign=0)
-        label.set_markup("<b>volctl settings</b>")
+        label.set_markup("<b>Volctl Settings</b>")
```
Comment (buzz) [inline volctl/prefs.py:33]: Why? volctl is the name, not Volctl.

File: data/apps.volctl.gschema.xml [hunk 2]
```diff
@@ -39,6 +39,11 @@
       <summary>OSD size</summary>
       <description>OSD scale in percent.</description>
     </key>
+    <key type="s" name="osd-position">
+      <default>"bottom-right"</default>
+      <summary>OSD position</summary>
+      <description>Select position for OSD volume; supported: center, top-right, bottom-right</description>
+    </key>
```
Comment (buzz) [inline data/apps.volctl.gschema.xml:46]: Should use gsettings schema enum.

Options: IMO at least all 4 corners + center of the screen should be supported.
Even better 9 positions: Top, TR, Right, BR, Bottom, BL, Left, TL and Center

File: volctl/meta.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 """Version string used by program and setup.py (using regex parsing)."""
 from gi.repository import Gtk
 
-VERSION = "0.9.0"
+VERSION = "0.10.0"
```
Comment (buzz) [inline volctl/meta.py:4]: Please don't bump the version.

File: volctl/osd.py [hunk 4]
```diff
@@ -83,7 +84,18 @@ def update_values(self, volume, mute):
...
+            self._move_to_center()
+        elif position == "bottom-right":
+            self._move_to_bottom_right()
+        elif position == "top-right":
+            self._move_to_top_right()
+        else:
+            print("Not supported osd-position. Supported are: center, bottom-right, top-right; osd-position=" + position)
+            raise ValueError
+
+    def _move_to_bottom_right(self):
```
Comment (buzz) [inline volctl/osd.py:98]: `_move_to_position` should be as concise and DRY as possible.

Is there any reason you don't use the `_compute_position` method? Make sure your code has proper multi screen support!

**Comment by raven2cz** [comment]
The points are fixed. 
I cannot simulate multi-screen support. It will be up to you. I tried to involve geometry of other screens by your guide and clean the coords calculation to clear understanding form.

**Comment by buzz** [comment]
> The points are fixed.
> I cannot simulate multi-screen support. It will be up to you. I tried to involve geometry of other screens by your guide and clean the coords calculation to clear understanding form.

A lot of people use multiple screens. So it's an important feature.  As I said, I can't make promises when I find the time to look at this. Let me know if you still want to take this on.

**Comment by raven2cz** [comment]
Ok, I will test it this week in more WMs and multiscreens.

**Comment by raven2cz** [comment]
I tested volctl in X11 WMs: dwm, qtile, awesomewm, xfce4, gnome, kde. Tests were with 2 monitors 4K and fullhd (different resolutions per screen). 
All 9 positions works correctly for each active screen 1 and 2. 
From my side, the tests are pass.

**Comment by raven2cz** [comment]
Unfunctional scale bar #51 this issue doesn't belong to this PR. It is independent behavior of pipewire. 

What is the status of PR? Can you merge it? I already test it all parts, see status from 16 days ago.

**Comment by buzz** [comment]
> Unfunctional scale bar #51 this issue doesn't belong to this PR. It is independent behavior of pipewire.
> 
> What is the status of PR? Can you merge it? I already test it all parts, see status from 16 days ago.

Sorry, @raven2cz, I've been out of office. Trying to fix to most urgent problems but I will come back to the OSD positions ASAP.
And thanks for the CPU load fix! :pray:

**Comment by buzz** [comment]
Merged in 5351aa65d0387a0c0828f4c3729e0408c125c280

**Comment by sysfu** [comment]
> I tested volctl in X11 WMs: dwm, qtile, awesomewm, xfce4, gnome, kde. Tests were with 2 monitors 4K and fullhd (different resolutions per screen). All 9 positions works correctly for each active screen 1 and 2. From my side, the tests are pass.

How did you get it working in dwm? When I try launching the volctl from the terminal I get this error

`─$ volctl
PulseAudio connected
Warning: Could not embed Gtk.StatusIcon.
Fatal error: Could not create a status icon. Are you sure you have a working notification area?
pulseaudio disconnected
`

**Pull request closed without merging**